tests/components/plex/test_config_flow.py

```diff
@@ -0,0 +1,265 @@
+"""Tests for Plex config flow."""
+from unittest.mock import MagicMock, Mock, patch, PropertyMock
+import plexapi.exceptions
+import requests.exceptions
+
+from homeassistant.components.plex import config_flow
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_TOKEN, CONF_URL
+
+from tests.common import mock_coro
+
+
+class MockAvailableServer:
+    """Mock avilable server objects."""
+
+    def __init__(self, name, client_id):
+        """Initialize the object."""
+        self.name = name
+        self.clientIdentifier = client_id
+        self.provides = ["server"]
+
+
+class MockConnection:
+    """Mock a single account resource connection object."""
+
+    def __init__(self):
+        """Initialize the object."""
+        self.httpuri = "http://1.2.3.4:32400"
+        self.uri = "http://4.3.2.1:32400"
+        self.local = True
+
+
+class MockConnections:
+    """Mock a list of resource connections."""
+
+    def __init__(self):
+        """Initialize the object."""
+        self.connections = [MockConnection()]
+
+
+async def test_bad_credentials(hass, aioclient_mock):
+    """Test when provided credentials are rejected."""
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": "user"}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+
+    with patch(
+        "plexapi.myplex.MyPlexAccount", side_effect=plexapi.exceptions.Unauthorized
+    ):
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={CONF_TOKEN: "12345"}
+        )
+
+        assert result["type"] == "form"
+        assert result["step_id"] == "user"
+        assert result["errors"]["base"] == "faulty_credentials"
+
+
+async def test_import_file_from_discovery(hass, aioclient_mock):
+    """Test importing a legacy file during discovery."""
+
+    mock_file_contents = {
+        "1.2.3.4:32400": {"ssl": False, "token": "12345", "verify": True}
+    }
+    file_host_and_port, file_config = list(mock_file_contents.items())[0]
+    used_url = f"http://{file_host_and_port}"
+
+    with patch("plexapi.server.PlexServer") as mock_plex_server, patch(
+        "homeassistant.components.plex.config_flow.load_json",
+        return_value=mock_coro(mock_file_contents),
```

Comment:
`load_json` isn't a coroutine but a regular function, so we shouldn't return a `mock_coro`, just the mock file contents.